enhancements/installer/openstack-no-fip-install.md

```diff
@@ -0,0 +1,135 @@
+---
+title: openstack-bring-your-own-external-connectivity
+authors:
+  - "@egarcia"
+  - "@adduarte"
+reviewers:
+  - "@mandre"
+  - "@pierre"
+approvers:
+  - "@mandre"
+  - "@pierre"
+creation-date: 2020-5-26
+last-updated: 2020-5-26
+status: implementable
+---
+
+# OpenStack BYO External Connectivity
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+Numerous customers have requested the ability to run the OpenShift on OpenStack IPI installer without using Floating IP addresses.
+In the installer, we use floating IPs as a way to set up external connectivity to the OpenShift cluster. Depending on how the OpenStack cluster
+is set up, the use of floating IPs may be impossible.
+
+## Motivation
+
+### Goals
+
+- Extend the installer to support provider networks
+- Allow the installer to be run without setting up external connectivity
+
+### Non-Goals
+
+This enhancement does not intend to provide any guidance or features that go further than allowing customers to deploy without Floating IPs. Operations like attaching loadbalancers to the entrypoints, or any day 2 operations are beyond the scope of this enhancement. It is important to note that while we aim to support installs without external connectivity, we are not supporting installs on networks that are unable to reach the host cluster's OpenStack APIs and services.
+
+## Proposal
+
+In order to implement this feature fully, the following changes must be made:
+1. Make `externalNetwork` an optional argument in the install-config when `machinesSubnet` is set
+2. Make setting  `lbFloatingIP` when no external network is passed enforced as invalid usage
```

Comment:
"invalid" -> "ignored"  ? do we really need  enforce or can we just ignore. I guess ignored would make it clearer.

Reply:
we could make the logic flow such that 
lbFloatingIP will only be looked for if externalNetwork is present, if externalNetwork is not present then lbFloatingIP is ignored, and thus preventing the FIP creation

Reply:
I would rather inconvenience them one time by enforcing proper usage, then potentially inconvenience them for a while because they are unsure what correct usage is.